### PR TITLE
Added katello installation test

### PR DESCRIPTION
**Pull request by lzap**
I know there is this `katello-deploy` tool, but this is very simplified
install-only version.

### CODE REVIEW

File: fb-install-katello.bats [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  if tFileExists /usr/sbin/firewalld; then
+    systemctl stop firewalld; systemctl disable firewalld
+  elif tCommandExists systemctl; then
+    systemctl stop iptables; systemctl disable iptables
+  else
+    service iptables stop; chkconfig iptables off
+  fi
+}
+
+@test "install yum utilities" {
```
Comment (ekohl) [inline fb-install-katello.bats]: Not just the yum utilities. Maybe just ```install utilities```?

File: fb-install-katello.bats [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  tPackageExists yum-utils || tPackageInstall yum-utils
+}
+
+@test "wait max 30 secs until network is up" {
+  ping -c1 -w30 8.8.8.8
+}
+
+@test "install SELinux tools" {
+    tPackageInstall install setools-console policycoreutils-python policycoreutils selinux-policy-devel
+    sepolgen-ifgen || true
```
Comment (ekohl) [inline fb-install-katello.bats]: Any reason this line is optional?

File: fb-install-katello.bats [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  tRHSubscribeAttach
+}
+
+@test "enable epel" {
+  tRHEnableEPEL
+}
+
+@test "configure repository" {
+  yum -y localinstall http://fedorapeople.org/groups/katello/releases/yum/nightly/katello/RHEL/${OS_VERSION}Server/x86_64/katello-repos-latest.rpm
+  yum -y localinstall http://yum.theforeman.org/nightly/el${OS_VERSION}/x86_64/foreman-release.rpm
```
Comment (ekohl) [inline fb-install-katello.bats]: You can use ```$FOREMAN_REPO``` here, right?

File: fb-install-katello.bats [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+@test "subscribe and attach channels" {
+  tRHSubscribeAttach
+}
+
+@test "enable epel" {
+  tRHEnableEPEL
+}
+
+@test "configure repository" {
+  yum -y localinstall http://fedorapeople.org/groups/katello/releases/yum/nightly/katello/RHEL/${OS_VERSION}Server/x86_64/katello-repos-latest.rpm
```
Comment (ekohl) [inline fb-install-katello.bats]: You can use ```$KATELLO_REPO``` here, right?

File: fb-install-katello.bats [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  yum -y localinstall http://yum.theforeman.org/nightly/el${OS_VERSION}/x86_64/foreman-release.rpm
+  yum -y localinstall http://yum.puppetlabs.com/puppetlabs-release-el-${OS_VERSION}.noarch.rpm
+  yum -y localinstall https://www.softwarecollections.org/en/scls/rhscl/v8314/epel-${OS_VERSION}-x86_64/download/rhscl-v8314-epel-${OS_VERSION}-x86_64.noarch.rpm
+  yum -y localinstall https://www.softwarecollections.org/en/scls/rhscl/ruby193/epel-${OS_VERSION}-x86_64/download/rhscl-ruby193-epel-${OS_VERSION}-x86_64.noarch.rpm
+}
+
+@test "install installer" {
+  tPackageExists katello || tPackageInstall katello || return $?
```
Comment (ekohl) [inline fb-install-katello.bats]: Any particular reason you need to manually return here?

**Comment by stbenjam** [comment]
Why the duplication?  Katello already has bats, what's the purpose of this?

I would think you should checkout katello-deploy and you can run a limited set of tests by running "katello-bats nightly" (there's others available too like content, and virt-whom)

**Comment by stbenjam** [comment]
And I really doubt foreman-installer is going to succeed with all the Katello extensions to foreman installed but not having run our installer.

**Comment by lzap** [comment]
All comments amended. @stbenjam yeah I know it exists, but I don't use vagrant and stuff, we have my virt-spawn script here which is based on `virt-builder` in this repo. This is easier for me than to rewrite my integration of my testing server.

**Comment by lzap** [comment]
Or I could write a test here which would download `katello-deploy` and run it... That would work too!

File: fb-install-katello.bats (hunk 2 above)
Comment (lzap) [inline fb-install-katello.bats]: > > +    sepolgen-ifgen || true
> 
> Any reason this line is optional?

Yeah, it does not always return 0.

-- 
Later,
 Lukas #lzap Zapletal

**Comment by lzap** [comment]
Amended. Also modified `virt-spawn` so it does not default to `fb-install-foreman`. You need to provide at least one script. If folks don't like this, I can remove this one. But otherwise I am not able to use `virt-spawn` to create a "clean" box (without any script executed).

**Comment by stbenjam** [comment]
>  Or I could write a test here which would download katello-deploy and run it... That would work too!

Cool, that works :+1:

**Comment by ekohl** [comment]
:+1:

**Comment by mmoll** [comment]
merged, thanks @lzap!

**Pull request merged**